server-common/src/main/java/org/apache/cassandra/sidecar/common/server/StorageOperations.java

```diff
@@ -102,4 +102,16 @@ default void outOfRangeDataCleanup(@NotNull String keyspace, @NotNull String tab
     {
         outOfRangeDataCleanup(keyspace, table, 1);
     }
+
+    /**
+     * @return the operation-mode of the Cassandra instance
+     */
+    String getOperationMode();
```

Comment:
not necessary for now. we could consider cache the operationMode in the future if it is queried a lot.

Reply:
I don't think we can cache the value because it can change during the lifecycle of the C* process. also this method should be `operationMode`

Reply:
Adding a bit more details to my comment. 

_Only if_ operationMode is queries a lot, it could cache the value and periodically refresh internally, to avoid loading Cassandra every single time.